### PR TITLE
Fix issue with force using of landscape mode in iOS games on macOS with M1

### LINKED ISSUE #5727: iOS games on macOS with M1: Fixed window size and portrait mode.
**Describe the bug (REQUIRED)**
Games that were published in App Store with macOS availability checkmark are launched in the window portrait mode, but the game thinks that it is in landscape mode. And there is no way to change the window size.

**To Reproduce (REQUIRED)**
Install any game made with Defold from the App Store on mac with M1. For example, [Look, Your Loot](https://apps.apple.com/ru/app/look-your-loot/id1339205191) or [Bring me Cakes](https://apps.apple.com/ru/app/bring-me-cakes-%D0%BB%D0%B0%D0%B1%D0%B8%D1%80%D0%B8%D0%BD%D1%82/id1167655230). 

Other games from the App Store can be resized to any window size. For example, [GoT Tale of Crows](https://apps.apple.com/us/app/game-of-thrones-tale-of-crows/id1483071787).

**Expected behavior (REQUIRED)**
The game launches with the landscaped window.

**Defold version (REQUIRED):**
Actual version for now is 1.2.180

**Platforms (REQUIRED):**
macOS
Big Sur
Macbook Air M1

**Screenshots (OPTIONAL):**
Screenshot 2021-04-07 at 14 57 37

### PR DESCRIPTION
**Pull request by AGulev**
Fixed issue when iOS game installed on M1 Mac forces to use landscape mode even if game made in portrait mode.

Fix https://github.com/defold/defold/issues/5727

### CODE REVIEW

File: engine/glfw/lib/ios/app/AppDelegate.m [hunk 1]
```diff
@@ -43,20 +43,6 @@ @implementation AppDelegate
 
 @synthesize window;
 
-- (void)forceDeviceOrientation;
```
Comment (AGulev) [inline engine/glfw/lib/ios/app/AppDelegate.m:46]: This is the reason of the bug.
It was added as a fix of[ the dev app on iOS 8.0](https://github.com/defold/defold/commit/29a654bfd30cf6ba991a3e5b4a954b3fd2d39258).
We don't have dev app anymore and user can create it's own or use the game as a dev app

[Later](https://github.com/defold/defold/commit/70e7531c93d7fd66657f5977e85c8c2e4c15a27e) it was moved as part of a workaround for iOS 8.0.0-8.0.2 which we don't support anymore. our min iOS version is 9.0

File: engine/glfw/lib/ios/app/AppDelegate.m [hunk 2]
```diff
@@ -78,14 +64,6 @@ - (void)reinit:(UIApplication *)application
         _glfwWin.height = tmp;
     }
 
-    [self forceDeviceOrientation];
-
-    float version = [[UIDevice currentDevice].systemVersion floatValue];
-    if (8.0 <= version && 8.1 > version) {
```
Comment (AGulev) [inline engine/glfw/lib/ios/app/AppDelegate.m:84]: Old workaround for iOS 8.0 - 8.1 But minimum supported iOS now is 9.0

File: engine/glfw/lib/ios/app/EAGLView.m [hunk 3]
```diff
@@ -14,18 +14,6 @@
...
- * Launching your Application in Landscape
- * https://developer.apple.com/library/ios/#technotes/tn2244/_index.html
- *
- * This is something we might want to add support for soon. The following is required:
- *   - Flip width/height when creating the window (in applicationDidFinishLaunching)
- *   - return (interfaceOrientation == UIInterfaceOrientationLandscapeRight);
- *     from shouldAutorotateToInterfaceOrientation
- *
- *   See TN2244 for more information
- */
```
Comment (AGulev) [inline engine/glfw/lib/ios/app/EAGLView.m:27]: Outdated information for old iOS versions

File: engine/glfw/lib/ios/app/ViewController.m [hunk 4]
```diff
@@ -35,23 +35,6 @@ - (void)dealloc
 - (void)createView:(BOOL)recreate
 {
     CGRect bounds = self.view.bounds;
-    float version = [[UIDevice currentDevice].systemVersion floatValue];
-    if (8.0 <= version && version < 8.1) {
```
Comment (AGulev) [inline engine/glfw/lib/ios/app/ViewController.m:39]: Old workaround for iOS 8.0 - 8.1 But minimum supported iOS now is 9.0

File: engine/glfw/lib/ios/app/ViewController.m [hunk 5]
```diff
@@ -90,29 +73,6 @@ - (void)viewWillDisappear:(BOOL)animated
     }
 }
 
-- (void)updateViewFramesWorkaround
-{
-    float version = [[UIDevice currentDevice].systemVersion floatValue];
-    if (8.0 <= version && version < 8.1) {
```
Comment (AGulev) [inline engine/glfw/lib/ios/app/ViewController.m:96]: Old workaround for iOS 8.0 - 8.1 But minimum supported iOS now is 9.0

File: engine/glfw/lib/ios/app/ViewController.m [hunk 6]
```diff
@@ -207,44 +167,8 @@ - (void)viewDidUnload
     [super viewDidUnload];
 }
 
-- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
-{
-    // NOTE: For iOS < 6
-    if (_glfwWin.portrait)
```
Comment (AGulev) [inline engine/glfw/lib/ios/app/ViewController.m:213]: Old workaround for iOS < 6. But minimum supported iOS now is 9.0

File: engine/glfw/lib/ios/app/ViewController.m [hunk 7]
```diff
@@ -207,44 +167,8 @@ - (void)viewDidUnload
...
-}
-
-- (void)didRotateFromInterfaceOrientation:(UIInterfaceOrientation)fromInterfaceOrientation
-{
-}
-
--(BOOL)shouldAutorotate{
-    // NOTE: Only for iOS6
-    return YES;
-}
```
Comment (AGulev) [inline engine/glfw/lib/ios/app/ViewController.m:232]: Old workaround for iOS < 6. But minimum supported iOS now is 9.0

File: engine/glfw/lib/ios/app/ViewController.m [hunk 8]
```diff
@@ -207,44 +167,8 @@ - (void)viewDidUnload
...
-
--(UIInterfaceOrientationMask)supportedInterfaceOrientations {
-    // NOTE: Only for iOS6
-    return UIInterfaceOrientationMaskLandscape | UIInterfaceOrientationMaskPortrait | UIInterfaceOrientationMaskPortraitUpsideDown;
-}
```
Comment (AGulev) [inline engine/glfw/lib/ios/app/ViewController.m:237]: Old workaround for iOS < 6. But minimum supported iOS now is 9.0

File: engine/glfw/lib/ios/app/ViewController.m [hunk 9]
```diff
@@ -207,44 +167,8 @@ - (void)viewDidUnload
...
-
 #pragma mark UIContentContainer
 
-// Introduced in iOS8.0
-- (void)viewWillTransitionToSize:(CGSize)size
-       withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator
-{
-    [self updateViewFramesWorkaround];
-}
```
Comment (AGulev) [inline engine/glfw/lib/ios/app/ViewController.m:246]: Old workaround for iOS 8.0 - 8.1 But minimum supported iOS now is 9.0

**Review by britzl — APPROVED** [review]
Nice cleanup!

File: engine/glfw/lib/ios/app/AppDelegate.m [hunk 10]
```diff
@@ -43,20 +43,6 @@ @implementation AppDelegate
 
 @synthesize window;
 
-- (void)forceDeviceOrientation;
-{
-    // Running iOS8, if we don't force a change in the device orientation then
-    // the framebuffer will not be created with the correct orientation.
-    UIDevice *device = [UIDevice currentDevice];
-    UIDeviceOrientation desired = UIDeviceOrientationLandscapeRight;
-    if (_glfwWin.portrait) {
```
Comment (AGulev) [inline engine/glfw/lib/ios/app/AppDelegate.m:52]: it doesn't work because _glfwWin.portrait set later than we use it here

**Pull request merged**